core/extension/gdextension_interface.cpp

```diff
@@ -904,8 +904,9 @@ static void gdextension_string_new_with_wide_chars(GDExtensionUninitializedStrin
 }
 
 static void gdextension_string_new_with_latin1_chars_and_len(GDExtensionUninitializedStringPtr r_dest, const char *p_contents, GDExtensionInt p_size) {
+	const size_t string_length = p_contents ? (p_size < 0 ? strlen(p_contents) : strnlen(p_contents, p_size)) : 0;
 	String *dest = memnew_placement(r_dest, String);
-	dest->append_latin1(Span(p_contents, p_contents ? _strlen_clipped(p_contents, p_size) : 0));
+	dest->append_latin1(Span(p_contents, string_length));
```

Comment:
When seeing it like this, this is kind of wasteful to create a span and call append if size is 0 (as append will do nothing)

Reply:
Yes, but we needn't worry about zero-length strings, I think. Those should be better caught statically, and adding another if check for those would unnecessarily burden non-trivial casese.

Reply:
yes, this definietely should not be the common case